awx/ui_next/src/screens/Template/shared/JobTemplateForm.jsx

```diff
@@ -104,6 +108,31 @@ class JobTemplateForm extends Component {
     }
   }
 
+  toggleCredentialSelection(newCredential) {
+    const { credentials: credentialsToUpdate } = this.state;
+    const { setFieldValue } = this.props;
+
+    let newCredentialsList;
+    const isSelectedCredentialInState =
+      credentialsToUpdate.filter(cred => cred.id === newCredential.id).length >
+      0;
+
+    if (isSelectedCredentialInState) {
+      newCredentialsList = credentialsToUpdate.filter(
+        cred => cred.id !== newCredential.id
+      );
+    } else {
+      newCredentialsList = credentialsToUpdate.filter(
+        credential =>
+          credential.kind === 'vault' || credential.kind !== newCredential.kind
+      );
+      newCredentialsList = [...newCredentialsList, newCredential];
```

Comment:
You can put all this logic in CredentialsLookup, then pass the full updated credentialsList to its `onChange` prop. Then in this component it's a simple `setFieldValue`. This will help keep the size of this form component down